test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go

```diff
@@ -68,48 +82,49 @@ var customLogsConfigTmplFile string
 //go:embed testfixtures/config.yaml.tmpl
 var configTmplFile string
 
-func multiFakeintakeStackDef(agentOptions ...agentparams.Option) *e2e.StackDefinition[multiFakeIntakeEnv] {
-	return e2e.EnvFactoryStackDef(func(ctx *pulumi.Context) (*multiFakeIntakeEnv, error) {
-		awsEnv, err := awsResources.NewEnvironment(ctx)
+func multiFakeIntakeAWS(agentOptions ...agentparams.Option) e2e.Provisioner {
+	runFunc := func(ctx *pulumi.Context, env *multiFakeIntakeEnv) error {
+		awsEnv, err := aws.NewEnvironment(ctx)
 		if err != nil {
-			return nil, err
+			return err
 		}
 
-		vm, err := ec2vm.NewEC2VMWithEnv(awsEnv)
+		host, err := ec2.NewVM(awsEnv, "nssfailover")
 		if err != nil {
-			return nil, err
+			return err
 		}
+		host.Export(ctx, &env.Host.HostOutput)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏Seems weird that creating any resource is followed by calling the Export function, I would expect for example a wrapper function to handle that transparently if it is really needed

(I know creating everything like we do in this test is uncommon so it's not a big deal honestly)

Reply:
It's not creating any resource, it's creating resources that need to be exported, hence the `Export`. In that case all resources need to be exported. It's not always the case.

Reply:
Hum ok